builtins/amp-img.js

```diff
@@ -89,6 +84,14 @@ export class AmpImg extends BaseElement {
     return promise;
   }
 
+  /** @override */
+  layoutCallback() {
+    // This is specifically needed for elements that does its work on prerender
+    // e.g. amp-img. This is only needed because currently prerendering
+    // undisplayed elements does not work. See issue #2742.
+    return this.updateImageSrc_();
```

Comment:
Depends on really how do we build this up, I don't anticipate this will be implemented by replacing the src of one single `img` element inside `amp-img` instead, they'll probably be 2 `img` elements one for prerendered state and one for laid out state. And we only hide (or destroy) the prerendered state once layout is complete.

The race condition can be mitigated in this setup by always using `prerenderCancelled` callback to have the final say on the prerendered DOM and either hide it or destroy it - Similar to what I did in `amp-instagram` and `amp-youtube` examples.